app/Authentication/src/Http/Controllers/SocialiteAuthController.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace App\Authentication\Http\Controllers;
+
+use Exception;
+use Knuckles\Scribe\Attributes\Group;
+use Knuckles\Scribe\Attributes\Endpoint;
+use Laravel\Socialite\Facades\Socialite;
+use Illuminate\Validation\ValidationException;
+use App\Authentication\Http\Resources\RedirectLinkResource;
+use App\Authentication\Http\Resources\AuthenticatedMemberResource;
+use Knuckles\Scribe\Attributes\UrlParam;
+
+class SocialiteAuthController
+{
+    #[Group('Authenticiation')]
+    #[Endpoint(title: 'Socialite Login Redirect', description: 'This endpoint redirect to the Social SignIn Form')]
+    #[UrlParam('provider', 'string', required: true, enum: ["github", "google", "linkedin"], example: "google")]
+    public function redirect($provider)
+    {
+        $this->validateProvider($provider);
+
+        return new RedirectLinkResource(Socialite::driver($provider)->stateless()->redirect()->getTargetUrl());
+    }
+
+    #[Group('Authenticiation')]
+    #[Endpoint(title: 'Socialite Login Callback', description: 'This endpoint sign in the users with Google, Github or LinkedIn Account')]
+    #[UrlParam('provider', 'string', required: true, enum: ["github", "google", "linkedin"], example: "google")]
+    public function handleCallback($provider)
+    {
+        $this->validateProvider($provider);
+
+        try {
+            $user = Socialite::driver($provider)->stateless()->user();
+
```

Comment:
Should we store some values in a database table so that we just retrieve the data and be able to match the email address if someone tried to register with it again normally with a password?

Reply:
Now when I think about it, we should probably throw an error if the user was not registered with a social provider and restrict these endpoints to be just for signing in rather than doing both, signing in and signing up.